packages/typespec-ts/test/modularIntegration/server.spec.ts

```diff
@@ -53,3 +55,83 @@ describe("Multiple Server Path Client", () => {
     }
   });
 });
+
+describe("NotVersioned Server Version Client", () => {
+  let client: NotVersionedClient;
+
+  beforeEach(() => {
+    client = new NotVersionedClient("http://localhost:3000", {
+      allowInsecureConnection: true,
+      retryOptions: {
+        maxRetries: 0
+      }
+    });
+  });
+
+  it("should work without apiVersion", async () => {
+    try {
+      const result = await client.withoutApiVersion();
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with param", async () => {
+    try {
+      const result = await client.withQueryApiVersion("v1.0");
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with path param", async () => {
+    try {
+      const result = await client.withPathApiVersion("v1.0");
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+});
+
+describe("Versioned Server Version Client", () => {
+  let client: VersionedClient;
+
+  beforeEach(() => {
+    client = new VersionedClient("http://localhost:3000", {
+      allowInsecureConnection: true,
+      retryOptions: {
+        maxRetries: 0
+      }
+    });
+  });
+
+  it("should work without apiVersion", async () => {
+    try {
+      const result = await client.withoutApiVersion();
+      assert.isUndefined(result);
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should work with param", async () => {
+    try {
+      const result = await client.withQueryApiVersion();
```

Comment:
in this case, query api-version parameter has become optional in the modular layer, we can skip passing it or pass it through options `{ apiVersion: "2022-12-01-preview" }`

Reply:
could we have a case for that? i believe it would failed with 400 in server side?

Reply:
I added a test case, but I don't really think that matters, we have set that value in modular layer, so it doesn't matter customer pass it or not if it's the same value.